### ISSUE TITLE
The number of parameters of a KAN layer in the paper?

### ISSUE THREAD

**Issue by hoangthangta** [yu-rp/KANbeFair#15]
**Number of Parameters of KAN and MLP**
 According to the paper: https://arxiv.org/pdf/2407.16674

We use d_in and d_out to denote the input and output dimensions of the neural network layer. We use K to denote the order of the spline, which corresponds to the parameter k of the official nn.Module KANLayer. It is the order of the Polynomial basis in the spline function. We use G to denote the number of spline intervals, which corresponds the num parameter of the official nn.Module
KANLayer. It is the number of intervals of the B-spline, before padding. It equals the number of control points - 1, before padding. After padding, there should be (K + G) functioning control points. 

**The learnable parameters in KAN include the B-spline control points, the shortcut weights, the B-spline weights, and the bias. The total learnable parameters are:**

`(d_in × d_out) × (G + K + 3) + d_out`

Sorry if I miss something. 

What I understand is the weight matrix only needs **(d_in × d_out) × (G + K)** parameters and the bias needs **d_out** parameters. How are other parameters such as the shortcut weights and B-spline control points? Why we add 3 here?

**Comment by XHX-HUST** [comment]
The following document provides a detailed record of my understanding and derivations regarding the calculation of parameter count and FLOPs in the KAN network architecture. Based on my derivations, I believe there may be some issues with the FLOPs calculation presented in the paper available at https://arxiv.org/pdf/2407.16674. I hope this document can be of some help to everyone. If there are any flaws in my derivations, please feel free to contact me promptly. Email: [haoxiangxu@hust.edu.cn](mailto:haoxiangxu@hust.edu.cn)

[参数与FLOPs.pdf](https://github.com/user-attachments/files/22416991/FLOPs.pdf)
